src/base/perturbations.jl

```diff
@@ -1,23 +1,25 @@
 abstract type Perturbation end
 
-struct ThreePhaseFault <: Perturbation
+mutable struct ThreePhaseFault <: Perturbation
     time::Float64
     Ybus::SparseMatrixCSC{Complex{Float64}, Int64}
 end
 
-get_affect(pert::ThreePhaseFault) =
+get_affect(::PSY.System, pert::ThreePhaseFault) =
```

Comment:
Do you need system here?

Reply:
not for the fault, but for reference changes yeah